corehq/apps/api/resources/v0_5.py

```diff
@@ -489,3 +493,71 @@ def dehydrate(self, bundle):
         bundle.data['product_name'] = bundle.obj.sql_product.name
         bundle.data['transaction_date'] = bundle.obj.report.date
         return bundle
+
+
+class UserDomain(object):
+
+    def __init__(self, domain_name='', project_name=''):
+        self.domain_name = domain_name
+        self.project_name = project_name
+
+
+class UserDomainsResource(Resource):
+    domain_name = fields.CharField(attribute='domain_name')
+    project_name = fields.CharField(attribute='project_name')
+
+    class Meta:
+        resource_name = 'user_domains'
+        authentication = ApiKeyAuthentication()
+        object_class = UserDomain
+        include_resource_uri = False
+
+    def obj_get_list(self, bundle, **kwargs):
+        return self.get_object_list(bundle.request)
+
+    def get_object_list(self, request):
+        couch_user = CouchUser.from_django_user(request.user)
+        results = []
+        for domain in couch_user.get_domains():
+            if not ZAPIER_INTEGRATION.enabled(domain):
+                continue
+            domain_object = Domain.get_by_name(domain)
+            results.append(UserDomain(
+                domain_name=domain_object.name,
+                project_name=domain_object.hr_name or domain_object.name
+            ))
+        return results
+
+
+class Form(object):
+
+    def __init__(self, form_xmlns='', form_name=''):
+        self.form_xmlns = form_xmlns
+        self.form_name = form_name
+
+
+class DomainForms(Resource):
+    form_xmlns = fields.CharField(attribute='form_xmlns')
+    form_name = fields.CharField(attribute='form_name')
+
+    class Meta:
+        resource_name = 'domain_forms'
+        authentication = ApiKeyAuthentication()
+        object_class = Form
+        include_resource_uri = False
+
+    def obj_get_list(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        couch_user = CouchUser.from_django_user(bundle.request.user)
+        if not ZAPIER_INTEGRATION.enabled(domain) or not couch_user.is_member_of(domain):
+            return []
+
+        results = []
+        for application in get_all_apps(domain):
+            forms_objects = application.get_forms(bare=False)
+            for form_object in forms_objects:
+                form = form_object['form']
+                module = form_object['module']
+                form_name = '{} > {} > {}'.format(application.name, module.name['en'], form.name['en'])
+                results.append(Form(form_xmlns=form.xmlns, form_name=form_name))
```

Comment:
i'm surprised there isn't a utility somewhere that already does this. could you potentially use `get_app_sources`?